underscore.js

```diff
@@ -520,19 +520,35 @@
     if (iteratee != null) iteratee = cb(iteratee, context);
     var result = [];
     var seen = [];
+    var seenPrimitive = {};
     for (var i = 0, length = array.length; i < length; i++) {
       var value = array[i],
           computed = iteratee ? iteratee(value, i, array) : value;
+
       if (isSorted) {
         if (!i || seen !== computed) result.push(value);
         seen = computed;
-      } else if (iteratee) {
-        if (!_.contains(seen, computed)) {
-          seen.push(computed);
+      } else {
+        var vType = typeof computed,
+          isPrimitive = _.contains(['number', 'string', 'boolean', 'undefined'], vType);
```

Comment:
@megawac See https://people.mozilla.org/~jorendorff/es6-draft.html#sec-properties-of-the-symbol-constructor. Each of the well-known symbols there changes how the object behaves in some way. It is not safe in a future with additional well-known symbols to modify these fields arbitrarily.